### PR TITLE
feat: support for https resource manager

### PR DESCRIPTION
**Pull request by irainia**
This PR is to add support for HTTP client request for Optimus resource manager.

### CODE REVIEW

File: ext/resourcemanager/optimus.go [hunk 1]
```diff
@@ -33,16 +35,43 @@ func NewOptimusResourceManager(resourceManagerConfig config.ResourceManager) (*O
...
+	if strings.HasPrefix(host, "https") {
+		certPool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("error reading system certificate: %w", err)
+		}
+
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:            certPool,
+				InsecureSkipVerify: true,
```
Comment (sbchaos) [inline ext/resourcemanager/optimus.go:67]: Can we remove `InsecureSkipVerify` from tlsConfig.

File: ext/resourcemanager/optimus.go [hunk 2]
```diff
@@ -33,16 +35,43 @@ func NewOptimusResourceManager(resourceManagerConfig config.ResourceManager) (*O
 	if err := mapstructure.Decode(resourceManagerConfig.Config, &conf); err != nil {
 		return nil, fmt.Errorf("error decoding resource manger config: %w", err)
 	}
+
 	if conf.Host == "" {
 		return nil, errors.New("optimus resource manager host is empty")
 	}
+
+	httpClient, err := newHTTPClient(conf.Host)
```
Comment (sbchaos) [inline ext/resourcemanager/optimus.go:43]: Can we also add the code to add http headers from config.

File: ext/resourcemanager/optimus.go (hunk 1 above)
Comment (irainia) [reply ext/resourcemanager/optimus.go:67]: sure, we can do that

File: ext/resourcemanager/optimus.go (hunk 2 above)
Comment (irainia) [reply ext/resourcemanager/optimus.go:43]: I checked that it wasn't possible here. But, we already did put the header under file `ext/resourcemanager/optimus.go` when constructing the request:

```go
for key, value := range o.config.Headers {
  // ...
}
```

File: ext/resourcemanager/optimus.go (hunk 1 above)
Comment (irainia) [reply ext/resourcemanager/optimus.go]: updated as suggested

**Review by sbchaos — APPROVED** [review]

**Pull request merged**